source/qb64.bas

```diff
@@ -1060,7 +1062,37 @@ IF C = 9 THEN 'run
 
     'execute program
 
+
+
+
     IF iderunmode = 1 THEN
+        IF NoExeSaved THEN
+            'This is the section which deals with if the user selected to run the program without
+            'saving an EXE file to the disk.
+            'We start off by first running the EXE, and then we delete it from the drive.
+            'making it a temporary file when all is said and done.
+            IF os$ = "WIN" THEN
+                SHELL QuotedFilename$(CHR$(34) + lastBinaryGenerated$ + CHR$(34)) + ModifyCOMMAND$ 'run the newly created program
+                SHELL _HIDE _DONTWAIT "del " + QuotedFilename$(CHR$(34) + lastBinaryGenerated$ + CHR$(34)) 'kill it

```

Comment:
IMO it would be worth making an issue that KILL can't handle files with spaces. Perhaps we should introduce a _DeleteFile function or something that can handle it properly, using SHELL for this is pretty ugly xD